### PR TITLE
FIX: align sklearnex `BasicStatistics._onedal_fit` with other algos

### PR DESCRIPTION
**Pull request by ethanglaser**
### Description 
Follow up to https://github.com/intel/scikit-learn-intelex/pull/1644. Specifically, aligns directly with https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/basic_statistics/incremental_basic_statistics.py#L178-L184

Fixes # - _issue number(s) if exists_

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.
- [ ] I have resolved any merge conflicts that might occur with the base branch.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [ ] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Comment by ethanglaser** [comment]
See fails in logs from infra PR 713, http://intel-ci.intel.com/ef3d6beb-40eb-f1c2-bbd9-a4bf010d0e2e

**Comment by ethanglaser** [comment]
fp32 infra PR will be merged after this - then we will automatically validate on other GPUs

**Review by icfaust — APPROVED** [review]
Thank you for the alignment, the ensure_2d thing is a bit interesting. Do we have test cases where a 1d array is passed?

**Review by samir-nasibli — APPROVED** [review]
@ethanglaser thank you very much for the alignment 👍 
Assuming green CI. Please share last internal CI run link

**Comment by olegkkruglov** [comment]
> Thank you for the alignment, the ensure_2d thing is a bit interesting. Do we have test cases where a 1d array is passed?

yes, we have. it must work with 1d arrays as well

**Comment by olegkkruglov** [comment]
by the way, exactly the same rows were in https://github.com/intel/scikit-learn-intelex/pull/1644 originally but were removed due to these comments.
https://github.com/intel/scikit-learn-intelex/pull/1644#discussion_r1655305869
https://github.com/intel/scikit-learn-intelex/pull/1644#discussion_r1658347359
@samir-nasibli @ethanglaser did I understand you wrong there?

**Comment by ethanglaser** [comment]
> Thank you for the alignment, the ensure_2d thing is a bit interesting. Do we have test cases where a 1d array is passed?

Yeah, https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/basic_statistics/tests/test_basic_statistics.py#L224 was failing until my latest commit (addition of ensure_2d=False)

**Comment by ethanglaser** [comment]
> by the way, exactly the same rows were in #1644 originally but were removed due to these comments. [#1644 (comment)](https://github.com/intel/scikit-learn-intelex/pull/1644#discussion_r1655305869) [#1644 (comment)](https://github.com/intel/scikit-learn-intelex/pull/1644#discussion_r1658347359) @samir-nasibli @ethanglaser did I understand you wrong there?

Oh oops thats my bad...
